File: Inventory-Management-System/src/main/java/com/brocode/entity/Product.java
package com.brocode.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

@Data
@Entity
@AllArgsConstructor
@NoArgsConstructor
@SuperBuilder
@EqualsAndHashCode(callSuper = true)
public class Product extends BaseEntity{
    private String productKey;
    private String name;
    private String description;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    private Category category;

    private Double price;
    private Double warningStockLevel;

    private Double inStock;
    private String unit;
}

File: mobile-application/app/src/main/java/com/example/inventory/data/Product.kt
package com.example.inventory.data

data class ProductResponseDto(
    val categoryId: Long,
    val categoryName: String,
    val productKey: String,
    val name: String,
    val description: String,
    val price: Double,
    val inStock: Double,
    val warningStockLevel: Double,
    val stockLevelLow: Boolean,
    val unit: String,
): BaseResponseDto()

data class ProductCreateDto(
    val categoryId: Long,
    val name: String,
    val description: String,
    val price: Double,
    val inStock: Double,
    val warningStockLevel: Double,
    val unit: String,
)